datasets/hkcancor/hkcancor.py

```diff
@@ -0,0 +1,161 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Hong Kong Cantonese Corpus (HKCanCor)."""
+
+from __future__ import absolute_import, division, print_function
+
+import os
+import xml.etree.ElementTree as ET
+
+import datasets
+
+
+_CITATION = """\
+@article{luke2015hong,
+  author={Luke, Kang-Kwong and Wong, May LY},
+  title={The Hong Kong Cantonese corpus: design and uses},
+  journal={Journal of Chinese Linguistics},
+  year={2015},
+  pages={309-330},
+  month={12}
+}
+"""
+
+_DESCRIPTION = """\
+The Hong Kong Cantonese Corpus (HKCanCor) comprise transcribed conversations
+recorded between March 1997 and August 1998. It contains recordings of
+spontaneous speech (51 texts) and radio programmes (42 texts),
+which involve 2 to 4 speakers, with 1 text of monologue.
+
+In total, the corpus contains around 230,000 Chinese words.
+The text is word-segmented, annotated with part-of-speech (POS) tags and
+romanised Cantonese pronunciation.
+
+Romanisation scheme - Linguistic Society of Hong Kong (LSHK)
+POS scheme - Peita-Fujitsu-Renmin Ribao (PRF) corpus (Duan et al., 2000)
+"""
+
+_HOMEPAGE = "http://compling.hss.ntu.edu.sg/hkcancor/"
+
+_LICENSE = "CC BY 4.0"
+
+_URL = "http://compling.hss.ntu.edu.sg/hkcancor/data/hkcancor-utf8.zip"
+
+
+class HkCanCor(datasets.GeneratorBasedBuilder):
+    """Hong Kong Cantonese Corpus (HKCanCor)."""
+
+    VERSION = datasets.Version("1.0.0")
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "conversation_id": datasets.Value("string"),
+                "speaker": datasets.Value("string"),
+                "turn_number": datasets.Value("int16"),
+                "words": datasets.Sequence(datasets.Value("string")),
+                "transcriptions": datasets.Sequence(datasets.Value("string")),
+                "pos_tags": datasets.Sequence(datasets.Value("string")),
```

Comment:
can you use the ClassLabel feature type here ?
You can find an example of usage in [conll2003](https://github.com/huggingface/datasets/blob/master/datasets/conll2003/conll2003.py) for example

Reply:
Done, thanks for the feedback @lhoestq!

I've also suggested some adjustments to the tagset. 

Basically, the original corpus included fine-grained tags that are (in my opinion) suited for qualitative linguistic analyses but challenging to use for NLP. To improve usability, I've left the original tags as-is, but also **added a mapping** that compresses the original set (n>100) to the **Universal Dependency 2.0** standard & into a **smaller set (n=16)**. This mapping is adapted from code from the PyCantonese library, which I have explicitly cited in the code and datasheet (both in the citation sections and where the dict was introduced). 

I know for this sprint the goal is to retain the properties of the original corpus with minimal additions, but I think it would be prohibitive to only have the original tags. Please let me know what you think. Thanks!